app/controllers/songs_controller.rb

```diff
@@ -12,7 +12,12 @@ def new
   end
 
   def create
+    artist = Artist.find_or_create_by(name: params[:song][:artist_name])
+    genre = Genre.find_or_create_by(id: params[:song][:genre_id])
```

Comment:
We wouldn't ever want to create something by an ID. You could just set the `genre_id` of a song the same way you set the title. Just include `genre_id` in your strong params and remove this line.